packages/now-build-utils/src/detect-builders.ts

```diff
@@ -70,28 +72,46 @@ function getFunctionBuilder(
 
   const src = (prevBuilder && prevBuilder.src) || file;
   const use = fn.runtime || (prevBuilder && prevBuilder.use);
-  const config: Config = Object.assign({}, prevBuilder && prevBuilder.config, {
-    functions,
-  });
 
-  if (!use) {
-    return prevBuilder;
+  const config: Config = { zeroConfig: true };
+
+  if (key) {
+    Object.assign(config, {
+      functions: {
+        [key]: fn,
+      },
+    });
   }
 
-  return { use, src, config };
+  return use ? { use, src, config } : prevBuilder;
 }
 
 async function detectFrontBuilder(
   pkg: PackageJson,
+  builders: Builder[],
   options: Options
 ): Promise<Builder> {
   for (const [dependency, builder] of getBuilders(options)) {
     const deps = Object.assign({}, pkg.dependencies, pkg.devDependencies);
-    const fnBuilder = getFunctionBuilder('package.json', builder, options);
 
     // Return the builder when a dependency matches
     if (deps[dependency]) {
-      return fnBuilder || builder;
+      if (options.functions) {
+        Object.entries(options.functions).forEach(([key, func]) => {
+          // When the builder is not used yet we'll use it for the frontend
```

Comment:
This might become slow if there are a lot of builders (B) and functions (F) because this will run B x F times.

What does `frontend` mean in your comment?

Reply:
The builder that is used for the frontend, like `@now/next` or `@now/static-build`